plaso/output/mediator.py

```diff
@@ -89,72 +89,74 @@ def GetFormattedMessages(self, event):
 
     return event_formatter.GetMessages(self._formatter_mediator, event)
 
-  def GetFormattedSources(self, event):
+  def GetFormattedSources(self, event, event_data):
     """Retrieves the formatted sources related to the event.
 
     Args:
       event (EventObject): event.
+      event_data (EventData): event data.
 
     Returns:
       tuple: containing:
 
         str: full source string or None if no event formatter was found.
         str: short source string or None if no event formatter was found.
     """
-    event_formatter = self.GetEventFormatter(event)
+    event_formatter = self.GetEventFormatter(event_data)
     if not event_formatter:
       return None, None
 
-    return event_formatter.GetSources(event)
+    return event_formatter.GetSources(event, event_data)
 
-  def GetFormatStringAttributeNames(self, event):
+  def GetFormatStringAttributeNames(self, event_data):
     """Retrieves the attribute names in the format string.
 
     Args:
-      event (EventObject): event.
+      event_data (EventData): event data.
 
     Returns:
       list[str]: list containing the attribute names. If no event formatter to
           match the event can be found the function returns None.
     """
-    event_formatter = self.GetEventFormatter(event)
+    event_formatter = self.GetEventFormatter(event_data)
     if not event_formatter:
       return None
 
     return event_formatter.GetFormatStringAttributeNames()
 
-  def GetHostname(self, event, default_hostname='-'):
+  def GetHostname(self, event_data, default_hostname='-'):
```

Comment:
Hostname is defined on event, not eventdata: https://github.com/log2timeline/plaso/blob/master/plaso/containers/events.py#L48 . See also comments on https://github.com/log2timeline/plaso/pull/2489

Reply:
See #2489 for explanation. I've also created https://github.com/log2timeline/plaso/pull/2533